lib/src/file_util.rs

```diff
@@ -347,6 +373,23 @@ mod platform {
         // note: if developer mode is not enabled the error code will be 1314,
         // ERROR_PRIVILEGE_NOT_HELD
 
+        // On Windows, "/" can't be part of valid file name, and "/" is also not a valid
+        // separator for the symlink target. See an example of this issue in
+        // https://github.com/jj-vcs/jj/issues/6934.
+        //
+        // We use debug_assert_* instead of assert_* because we want to avoid panic in
+        // release build, and we are sure that we shouldn't create invalid symlinks in
+        // tests.
+        debug_assert_ne!(
+            original
+                .as_ref()
+                .as_os_str()
+                .to_str()
+                .map(|path| path.contains("/")),
+            Some(true),
+            "Invalid symlink target: {}.",
+            original.as_ref().display()
+        );
```

Comment:
I wouldn't add this assertion because it's not wrong to create a dead symlink. It's up to caller whether to interpret symlink target as a disk path.

Reply:
> If we do allow the symlink to store arbitrary bytes,

I think `&Path` or `&OsStr` is good because Windows API wouldn't accept arbitrary bytes, but that's minor detail. (Unix version wouldn't also work if the `original` contained NUL bytes.)

> Nevertheless, we all agree that, in working copy, we shouldn't create a definitely invalid symlink in test, so I think we are all Ok with leaving the assertion in working copy

I don't think it makes sense to assert what `symlink_target_convert_to_disk()` would do immediately after the function call, but I'm not strongly against the assertion, yes.